esphome/components/mitsubishi/mitsubishi.cpp

```diff
@@ -22,37 +40,211 @@ const uint16_t MITSUBISHI_HEADER_MARK = 3500;
 const uint16_t MITSUBISHI_HEADER_SPACE = 1700;
 const uint16_t MITSUBISHI_MIN_GAP = 17500;
 
+// Marker bytes
+const uint8_t MITSUBISHI_BYTE00 = 0X23;
+const uint8_t MITSUBISHI_BYTE01 = 0XCB;
+const uint8_t MITSUBISHI_BYTE02 = 0X26;
+const uint8_t MITSUBISHI_BYTE03 = 0X01;
+const uint8_t MITSUBISHI_BYTE04 = 0X00;
+const uint8_t MITSUBISHI_BYTE13 = 0X00;
+const uint8_t MITSUBISHI_BYTE16 = 0X00;
+
+climate::ClimateTraits MitsubishiClimate::traits() {
+  auto traits = climate::ClimateTraits();
+  traits.set_supports_action(false);
+  traits.set_visual_min_temperature(MITSUBISHI_TEMP_MIN);
+  traits.set_visual_max_temperature(MITSUBISHI_TEMP_MAX);
+  traits.set_visual_temperature_step(1.0f);
+  traits.set_supported_modes({climate::CLIMATE_MODE_OFF});
+
+  if (this->supports_cool_)
+    traits.add_supported_mode(climate::CLIMATE_MODE_COOL);
+  if (this->supports_heat_)
+    traits.add_supported_mode(climate::CLIMATE_MODE_HEAT);
+
```

Comment:
```suggestion
  if (this->supports_cool_ && this->supports_heat_)
    traits.add_supported_mode(climate::CLIMATE_MODE_HEAT_COOL);
```
Adding support for heat_cool

Reply:
Right. I guess we are just in the dumbness of the ClimateIR's interface. On one side, it splits each feature in unique name "support_X", but it creates this strange combined mode "heat & cool" that you have to combine yourself. I guess it should be called "auto" and the logic move would have been to have `supports_auto` instead. There is no reason why if you support heat & cool that you should support auto mode. 

The issue here is that we can't unsupport auto mode in current mitsubishi code, since all modes enable the auto mode albeit in a strange fashion, setting to dry will auto set the temperature to 24°C and so on. 

I think you're right here that we should add this, but in a later PR this should be changed in ClimateIR component.

Reply:
Thanks. 
Somehow we should allow heat_cool and this is the only way that I can think of right now. 
Yes, agreed, on the rest of your comments. I have thought the same thing(s).